### PR TITLE
log backup: Refactor advancer

### LINKED ISSUE #38152: br/log-bakcup: make full scan using the tick interface to retry
## Enhancement
For now, we sleep directly in the `full scan` stage. Which is hard to debugging and dirty. We can refactor it to make the retry state into the tick state, then we can retry at each tick: this would also make testing easier and isolate the algorithm part and the real world I/O.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #38152

Problem Summary:
Before we would try to finish phase 1 (Get all checkpoints) in one tick.
Which is strange.

### What is changed and how it works?
This PR has refactored those logic, making it tries to request the range when ticking.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Have tested it in TPCC workload.
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/streamhelper/config/advancer_conf.go [hunk 1]
```diff
@@ -9,17 +9,22 @@ import (
...
-	flagBackoffTime      = "backoff-time"
-	flagTickInterval     = "tick-interval"
-	flagFullScanDiffTick = "full-scan-tick"
-	flagAdvancingByCache = "advancing-by-cache"
+	flagBackoffTime          = "backoff-time"
+	flagMaxBackoffTime       = "max-backoff-time"
+	flagTickInterval         = "tick-interval"
+	flagFullScanDiffTick     = "full-scan-tick"
+	flagAdvancingByCache     = "advancing-by-cache"
+	flagConsistencyCheckTick = "consistency-check-tick"
```
Comment (3pointer) [inline br/pkg/streamhelper/config/advancer_conf.go:17]: maybe we should check these config name with @IANTHEREAL

Comment (YuJuncen) [reply br/pkg/streamhelper/config/advancer_conf.go:17]: Those configs aren't exported. 🤔

**Comment by YuJuncen** [comment]
I have some new idea about doing this. Let's close this firstly. 🤔

**Pull request closed without merging**